### ISSUE TITLE
review latest WCMP2 draft (2.0.0-alpha2)

### ISSUE THREAD

**Issue by tomkralidis** [wmo-im/wcmp2#22]
@wmo-im/tt-wismd as discussed, this issue to track comments and change requests for 2.0.0-alpha2.

The document is attached here for access: [wcmp2.pdf](https://github.com/wmo-im/wcmp2/files/8991224/wcmp2.pdf)

**Comment by tomkralidis** [comment]
- 7.1.11 mentions to specify at least 2 contacts (main point of contact and originator).  What if they are identical (duplication)?
- We should add requirements for granularity and parent/child relationships

**Comment by josusky** [comment]
Hm, the paragraph at the beginning of the chapter reads "A WCMP record can be **one or more** providers ...". First of all there is a typo ("can have" seems more appropriate to me), secondly it contradicts the _requirement 10_, that calls for "at least TWO". Now when it comes to the duplication, couldn't we solve it by allowing more than one "role" per provider?
@tomkralidis I am not sure if your comment about "parent/child relationships" is generic, or relates to the providers. But we can discuss this during today's teleconference.

**Comment by tomkralidis** [comment]
TT-WISMD 2022-06-29:
- 7.1.9: show bbox values as floats
- changes:
  - comments here
  - small issues/typos/etc. : add as pull request
  - 7.1.16
    - s/provider-specific context/provider-specific details/
  - topic linkages:
    - WIS topic (category) && MQTT topic (actionable links/infra, from Global Broker)
      - will typically be consistent
      - clarify in document
  - 7.1.13
    - DOI should be a link, not only an identifier
 - 7.1.8
   - should we explicitly say "WIS 2 topic hierachy"?

**Comment by tomkralidis** [comment]
@wmo-im/tt-wismd please see #27 which addresses most of the comments above.

NOTE: addressing 7.1.9 (extents refactor) and adding a recommendation on granularity will be published as their own dedicated PRs.

**Comment by tomkralidis** [comment]
Good point @jsieland.  We could align [Req 2](https://github.com/wmo-im/wcmp2/blob/main/standard/requirements/core/REQ_identifier.adoc) with the [WIS 2 topic structure](https://github.com/wmo-im/wis2-topic-hierarchy#wis2-topic-hierarchy) as follows:

- A WCMP record identifier SHALL have the following notation: `urn:x-wmo:md:{country}:{center_id}:{local_identifier}`
- A WCMP record identifier SHALL contain a `country` based on the associated vocabulary as specified in the WIS 2 topic hierarchy
- A WCMP record identifier SHALL contain a `center_id` based on the associated vocabulary as specified in the WIS 2 topic hierarchy

Example:

```json
"id": "urn:x-wmo:md:CAN:eccc-msc:observations.swob"
```

**Comment by gaubert** [comment]
Hi @tomkralidis and @wmo-im/tt-wismd 

Please find in the attachment my review from the PDF document as it was more convenient to go through it in that way and see the overall result. I believe we should add some clarification linked to our domain in some areas and added some comments and suggested changes for that. Let me know what you think.

[wcmp2.pdf](https://github.com/wmo-im/wcmp2/files/9035085/wcmp2.pdf)

**Comment by jsieland** [comment]
@gaubert 
To me it seems like `Recommendation 8: /rec/core/links` might be an old one from before the rewrite of the data policy? If not I also do not understand what this REC is about.

**Comment by jsieland** [comment]
Second review from myself and also including some additional comments from @antje-s 

[wcmp2_AS_JS.pdf](https://github.com/wmo-im/wcmp2/files/9053156/wcmp2_AS_JS.pdf)

**Comment by tomkralidis** [comment]
> @gaubert To me it seems like `Recommendation 8: /rec/core/links` might be an old one from before the rewrite of the data policy? If not I also do not understand what this REC is about.

I think this is since removed?

**Comment by tomkralidis** [comment]
> Good point @jsieland. We could align [Req 2](https://github.com/wmo-im/wcmp2/blob/main/standard/requirements/core/REQ_identifier.adoc) with the [WIS 2 topic structure](https://github.com/wmo-im/wis2-topic-hierarchy#wis2-topic-hierarchy) as follows:
> 
>     * A WCMP record identifier SHALL have the following notation: `urn:x-wmo:md:{country}:{center_id}:{local_identifier}`
> 
>     * A WCMP record identifier SHALL contain a `country` based on the associated vocabulary as specified in the WIS 2 topic hierarchy
> 
>     * A WCMP record identifier SHALL contain a `center_id` based on the associated vocabulary as specified in the WIS 2 topic hierarchy
> 
> 
> Example:
> 
> ```json
> "id": "urn:x-wmo:md:CAN:eccc-msc:observations.swob"
> ```
> 
> Suggestions/ideas/feedback?

@jsieland / all: FYI PR #39 also includes updates based on the above, for review/feedback.

**Comment by tomkralidis** [comment]
> Second review from myself and also including some additional comments from @antje-s
> 
> [wcmp2_AS_JS.pdf](https://github.com/wmo-im/wcmp2/files/9053156/wcmp2_AS_JS.pdf)

Thanks @jsieland.  Updates based on your comments also in PR #39. Some comments for feedback/discussion:
- GDC will support ONLY WCMP2, but that detail should be in the GDC documentation
- `properties.type`: there is a TODO for WCMP2 codelist definition
- access control examples are covered in PR #20 
- for MQTT links, per https://www.iana.org/assignments/link-relations/link-relations.xhtml, we could change the preferred link relation from `data` to `current` (refers to a resource containing the most recent item(s) in a collection of resources [RFC5005])


current 	Refers to a resource containing the most recent item(s) in a collection of resources. 	[[RFC5005](https://www.iana.org/go/rfc5005)]

**Comment by amilan17** [comment]
1.  Add requirement to 7.1.8 (themes and topic hierarchy): A WCMP record SHALL provide a properties.themes.scheme property.
2.  Providers: I think there should be 3 types of contacts required. point of contact for the distrubution of the dataset and metadata contact for the creation of the record. The other provider should be the originator of the dataset, but in my experience from working at an archive, the originator is no longer "in charge" of the dataset and it was often included without contact information. 
3.  Remove reference to services for Properties Type?

**Comment by amilan17** [comment]
Consider making the following more consistent (review based on 2.0.0-alpha2 document): 

**the resource**

*   resource (~23)
*   dataset (~15)
*   data (~30, data is hard wired into a lot of content that we cannot change, e.g. Spatial Data on the Web…)
*   dataset’s data is … (1)
*   product (~15)

**the metadata record**

1.  WCMP record (78)
2.  WCMP discovery metadata record/s 
3.  metadata record/s
4.  discovery metadata record/s

**specification vs standard --** I think we are constently refererring to WCMP as a standard and OARec as a specification, but in the end this will become a standard too. So should we change the language now, so we don't forget later?

**Comment by amilan17** [comment]
Editorial recommendations: 

1.  Add summary table listing all the properties and their requirements mandatory, optional, conditional for a quick overview of everything
2.  Move examples after the requirement tables
3.  Provide real examples (e.g. NOT fake-doi, example.org…)
4.  Write out 1..n ==> one to many

**Comment by amilan17** [comment]
*    [ ]  change [https://www.wmo.int/spec/wcmp/2.0/conf/core](https://www.wmo.int/spec/wcmp/2.0/conf/core) ==> https://schemas.wmo.int/

**Comment by tomkralidis** [comment]
@amilan17 actually that value is a URI, and expected.

**Comment by jsieland** [comment]
**tl;dr**
While working on writing examples following WCMP2 I think the given information and examples for writing metadata are useful but the spec needs a bit more clarification in some parts.


**The long story**
As part of reviewing/producing examples I wrote something like a "working template cheat sheet" kinda thing for myself using the WCMP2 spec from begin to end.

During the process I came across the following points - many of them may be more or less self-explanatory if you are familiar with the topics at hand but could be harder to understand for others:
- [x] `properties.type` was missing in table 2
- [ ] `type` as per /req/core/validation (REQ-1-C) is missing in table 2
- [ ] hard to keep in mind what is included in `properties` and what not, e.g.:
	- 7.1.5. Conformance -> not in `properties`
	- 7.1.6. Properties Type -> in `properties` and label indicates difference to `type` from GeoJSON (see 7.1.2. Validation)
	- 7.1.7. Title -> in `properties` but at first glance it seems like at the same level as conformance
	- ...
	- 7.1.11. Geospatial and Temporal Extents
	    - 7.1.11.1. Geospatial Extent
	        - `geometry` -> not in `properties`, same level as conformance 
	        - `extent.spatial` -> in `properties`, same level as properties.type and title etc.
	    - Temporal Extent
	        - `time` -> not in `properties`, same level as conformance 
	        - `extent.temporal` -> in `properties`, same level as properties.type and title etc.
- [ ] "Temporal Extent" 
	- seems to have the wrong header hierarchy -> should be 7.1.11.3. Temporal Extent with 7.1.11.4. Additional Temporal Extents
- [ ] 7.1.12. Providers: 
	- Question: /req/core/providers (REQ-10-B): "[...] Providers are defined as either a URI or inline." -> This sencence is based on the "oneOf" in https://github.com/opengeospatial/ogcapi-records/blob/master/core/openapi/schemas/party.yaml? How to include the roles with using the URI option?	
- [ ] 7.1.18. Distribution Information: 
	- REQ/REC/PER are listed in 7.1.18.2. Examples (?)
	- this chapter has extensive examples (which is good) ... but it took some time to see which example is covered by which REQ/REC/PER, it could be helpful to have them more together or have a reference which REQ/REC/PER an example is about
	- Question: /req/core/links (REQ-15-C) and /rec/core/links (REC-9-A) are linked? Meaning REQ-15-C should be indicated as stated in REC-9-A?
	- Question: in the last example there is a `security` property, which is also mentioned in "A.2. Links" but not in /req/core/links (REQ-15-E)?
- [ ] in some subchapters the examples are given before the actual requirements/recommendations/permissions which feels kinda odd for long examples 
- [ ] Question: is there any information on adding additional properties or did I just overlooked it?

A lot of useful information is given in the spec but it feels a bit unstructured in the parts mentioned above. I assume this just comes from the working process and that we know what we want to say - but for lowering the barrier it could be helpful to clarify some points.

_Note: I'm out of office until 2022-10-05, just in case there are any questions or further discussions needed :)_

**Comment by amilan17** [comment]
@jsieland -- you said "hard to keep in mind what is included in `properties` and what not..." -- do you think that all the fields under "properties" should be grouped together?

**Comment by amilan17** [comment]
"in some subchapters the examples are given before the actual requirements/recommendations/permissions which feels kinda odd for long examples"

@jsieland -- I also find this awkard and created an issue for it. See #69

**Comment by jsieland** [comment]
> @jsieland -- you said "hard to keep in mind what is included in `properties` and what not..." -- do you think that all the fields under "properties" should be grouped together?

Yes, I think it would be helpful. Either by indicating in the chapter description (like 7.1.6. Properties Type) or by grouping chapters.

Especially because `geometry` and `time` are mandatory but are in a deeper nested chapter: that's is easy to forget. Maybe for those two we can put them in extra subchapters 7.1.x Geometry and 7.1.x Time and have 7.1.x Additional Geospatial and Temporal Extents (which include `extent.spatial` and `extent.temporal`).

**Comment by tomkralidis** [comment]
Thanks @jsieland

- 7.1.12: I've opened https://github.com/opengeospatial/ogcapi-records/issues/187 to help address roles in provider URLs
- PR #70 attempts to address most of the comments, for review/comment (note that a PDF build can be found in the PR to review 'in context').  The PR also addresses:
  - #38
  - #67
  - #68
  - #69
